isthmus/src/main/java/io/substrait/isthmus/expression/WindowFunctionConverter.java

```diff
@@ -4,172 +4,98 @@
 import org.apache.calcite.rex.RexWindowBound;
+import org.apache.calcite.sql.type.SqlTypeName;
 
 public class WindowFunctionConverter
     extends FunctionConverter<
         SimpleExtension.WindowFunctionVariant,
-        WindowFunctionInvocation,
-        WindowFunctionConverter.WrappedAggregateCall> {
-
-  private AggregateFunctionConverter aggregateFunctionConverter;
+        Expression.WindowFunctionInvocation,
+        WindowFunctionConverter.WrappedWindowCall> {
 
   @Override
   protected ImmutableList<FunctionMappings.Sig> getSigs() {
     return FunctionMappings.WINDOW_SIGS;
   }
 
   public WindowFunctionConverter(
-      List<SimpleExtension.WindowFunctionVariant> functions,
-      RelDataTypeFactory typeFactory,
-      AggregateFunctionConverter aggregateFunctionConverter,
-      TypeConverter typeConverter) {
+      List<SimpleExtension.WindowFunctionVariant> functions, RelDataTypeFactory typeFactory) {
     super(functions, typeFactory);
-    this.aggregateFunctionConverter = aggregateFunctionConverter;
   }
 
   @Override
-  protected WindowFunctionInvocation generateBinding(
-      WrappedAggregateCall call,
+  protected Expression.WindowFunctionInvocation generateBinding(
+      WrappedWindowCall call,
       SimpleExtension.WindowFunctionVariant function,
       List<FunctionArg> arguments,
       Type outputType) {
-    AggregateCall agg = call.getUnderlying();
+    RexOver over = call.over;
+    RexWindow window = over.getWindow();
+
+    var partitionExprs =
+        window.partitionKeys.stream()
+            .map(r -> r.accept(call.rexExpressionConverter))
+            .collect(java.util.stream.Collectors.toList());
 
     List<Expression.SortField> sorts =
-        agg.getCollation() != null
-            ? agg.getCollation().getFieldCollations().stream()
-                .map(r -> SubstraitRelVisitor.toSortField(r, call.inputType))
+        window.orderKeys != null
+            ? window.orderKeys.stream()
+                .map(rfc -> toSortField(rfc, call.rexExpressionConverter))
                 .collect(java.util.stream.Collectors.toList())
             : Collections.emptyList();
     Expression.AggregationInvocation invocation =
-        agg.isDistinct()
+        over.isDistinct()
             ? Expression.AggregationInvocation.DISTINCT
             : Expression.AggregationInvocation.ALL;
+
+    WindowBound lowerBound = toWindowBound(window.getLowerBound(), call.rexExpressionConverter);
+    WindowBound upperBound = toWindowBound(window.getUpperBound(), call.rexExpressionConverter);
+
     return ExpressionCreator.windowFunction(
         function,
         outputType,
         Expression.AggregationPhase.INITIAL_TO_RESULT,
         sorts,
         invocation,
+        partitionExprs,
+        lowerBound,
+        upperBound,
         arguments);
   }
 
-  public Expression.Window convert(
-      RelNode input,
-      Type.Struct inputType,
+  public Optional<Expression.WindowFunctionInvocation> convert(
```

Comment:
```java
public Optional<Expression.WindowFunctionInvocation> convert
```

This output types brings `convert` inline with both ScalarFunctionConverter and AggregateFunctionConverter.